aee.js

```diff
@@ -23,3 +23,15 @@ Array.prototype.any = function (spec) {
   }
   return false
 }
+
+Array.prototype.take = function (howMany, spec) {
+  let result = []
+  if (howMany <= 0) return result
+  const len = this.length
+  const searchSpec = (typeof spec === 'function') ? spec : () => true
```

Comment:
I'm not quite sure I did understand a single bit of what you said, the only I can add is that `const name = value` means that you can't change the reference it points to or the value it contains. In case it points to an object you can change it's inner props or content, that way you can push, pop, etc into/from a `const result = []`